narwhal/worker/src/handlers.rs

```diff
@@ -170,24 +177,23 @@ impl<V: TransactionValidator> PrimaryToWorker for PrimaryReceiverHandler<V> {
 
         // Keep attempting to retrieve missing batches until we get them all or the client
         // abandons the RPC.
+        // TODO: synchronize() should only be used for header payloads, so remove the broadcast
+        // logic after first attempt.
+        // TODO: synchronize() should not be cancelled when the higher level RequestVote gets
+        // cancelled, if payload requests are already inflight. This avoids wasting work and
```

Comment:
Maybe it is okay to cancel header synchronize if request vote gets cancelled? Only a guarantee we will need the payload when it becomes part of a certificate.

Reply:
I think header payload should be fetched with its own timeout, e.g. 5s ~ 10s. RequestVote gets cancelled much more frequently, e.g. 0.5s ~ 1s. I think worker should not waste work. But this is not a major optimization.